src/client/api/event.c

```diff
@@ -112,6 +113,8 @@ daos_eq_lib_reset_after_fork(void)
 {
 	eq_ref            = 0;
 	ev_thpriv_is_init = false;
+	crt_reset_lock();
+	D_MUTEX_INIT(&daos_eq_lock, NULL);
```

Comment:
this looks a bit hacky. why is this needed?

Reply:
This will clear the lock status in the new child process after fork(). It should be safe to do so since it does not affect parent process. I observed hang of waiting for lock in my previous local tests.

Reply:
hmm does it mean you fork while the lock is acquired? how do we get into that situation, like what is the use case? I understand if you call fork() then child tries to access an acquired mutex by the parent, it will hang forever, but are we really doing that?
if fork() then exec(), we don't need to reinitialize? 
how do we guarantee there aren't other locks that need to be reinitialized?